src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -2141,6 +2141,11 @@ public void addLocalApplicationState(ApplicationState applicationState, Versione
         addLocalApplicationStates(Arrays.asList(Pair.create(applicationState, value)));
     }
 
+    public void addLocalApplicationState(ApplicationState state1, VersionedValue value1, ApplicationState state2, VersionedValue value2)
```

Comment:
added this method as the locking is done at the call level, so when we have STATUS and STATUS_AND_PORT as 2 different calls we can get into a state where we gossip one changed and the other unchanged; by using this new method both status are updated in the same lock

Reply:
was this an actual problem? STATUS is deprecated in 4.0 and not supposed to be used in 5.0 (that probably needs its own ticket though)

Reply:
> was this an actual problem? 

its 100% unrelated... I tried solving this a different way but moved away from it (add a status_hash app state, so we can know if the cluster is in-sync with a change made locally).  I can force simulate an issue, but I have not heard a real problem in the wild yet.

Reply:
with STATUS deprecated, I don't think there should be a problem as long as STATUS_AND_PORT is set.

Reply:
I can remove from this patch